torchtune/utils/memory.py

```diff
@@ -20,19 +20,27 @@
 
 _log: logging.Logger = get_logger()
 
+ACWrapPolicyType: Type = Union[Set[Type], Callable[[nn.Module, bool, int], bool]]
+
 
 def set_activation_checkpointing(
-    model: nn.Module, auto_wrap_policy: Optional[Set[nn.Module]] = None, **kwargs
+    model: nn.Module, auto_wrap_policy: ACWrapPolicyType, **kwargs
 ) -> None:
-    """Utility to setup activation checkpointing and wrap the model for checkpointing.
+    """Utility to apply activation checkpointing to the passed in model.
 
     Args:
         model (nn.Module): Model to setup activation checkpointing.
-        auto_wrap_policy (Optional[Set[nn.Module]]): Policy to wrap module.
-        **kwargs: additional arguments to pass to torch.distributed activation checkpointing.
+        auto_wrap_policy (ACWrapPolicyType): Policy to wrap module.
+            This can either be a set of ``nn.Module`` types, in which case, modules of the specified type(s)
+            will be wrapped individually with activation checkpointing or a ``callable`` policy describing
```

Comment:
nit: commas
```suggestion
            This can either be a set of ``nn.Module`` types, in which case modules of the specified type(s)
            will be wrapped individually with activation checkpointing, or a ``callable`` policy describing
```